api/internal/managers/app/config/config.go

```diff
@@ -1,13 +1,82 @@
 package config
 
 import (
-	"context"
+	"fmt"
+
+	kotsv1beta1 "github.com/replicatedhq/kotskinds/apis/kots/v1beta1"
+	"github.com/replicatedhq/kotskinds/multitype"
+	"github.com/tiendc/go-deepcopy"
 )
 
+func (m *appConfigManager) GetConfig(config kotsv1beta1.Config) (kotsv1beta1.Config, error) {
+	return filterAppConfig(config)
+}
+
 func (m *appConfigManager) GetConfigValues() (map[string]string, error) {
 	return m.appConfigStore.GetConfigValues()
 }
 
-func (m *appConfigManager) SetConfigValues(ctx context.Context, values map[string]string) error {
-	return m.appConfigStore.SetConfigValues(values)
+func (m *appConfigManager) SetConfigValues(config kotsv1beta1.Config, configValues map[string]string) error {
+	filteredValues := make(map[string]string)
+
+	// only include values for enabled groups and items, otherwise keep the original value
+	for _, g := range config.Spec.Groups {
+		for _, i := range g.Items {
+			if !isItemEnabled(g.When) || !isItemEnabled(i.When) {
+				filteredValues[i.Name] = i.Value.String()
```

Comment:
wouldn't this be applying the value from the config to config values and potentially be adding values the user never submitted? i think we should add to filtered values if item is enabled, but not add at all if not, no?